src/Agent.Worker/ExpressionManager.cs

```diff
@@ -66,6 +85,11 @@ protected sealed override object EvaluateCore(DT.EvaluationContext evaluationCon
             {
                 var executionContext = evaluationContext.State as IExecutionContext;
                 ArgUtil.NotNull(executionContext, nameof(executionContext));
+                if (executionContext.CancellationToken.IsCancellationRequested)
```

Comment:
This is messed up I think. It looks like the step execution context is passed in, but since the tokens are not linked anymore (with this PR) then I think this should check the job execution context (parent of one passed in).